rosidl_runtime_cpp/include/rosidl_runtime_cpp/traits.hpp

```diff
@@ -130,7 +130,7 @@ void value_to_yaml(const std::u16string & value, std::ostream & out)
   auto flags = out.flags();
   size_t index = 0;
   while (index < value.size()) {
-    int16_t character = *reinterpret_cast<const int16_t *>(&value[index]);
+    int16_t character = static_cast<int16_t>(value[index]);
```

Comment:
```suggestion
    int16_t character = static_cast<uint_least16_t>(value[index]);
```

Reply:
Applied in 63bf202799dff3d24aebbca55b2c49a0acb5214a.